yocto-base/00_flash_and_verify.robot

```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Library             Collections
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../../sonoff-rest-api/sonoff-api.robot
+Resource            ../../rtectrl-rest-api/rtectrl.robot
+Resource            ../../variables.robot
+Resource            ../../keywords.robot
+Resource            ../lib/sd-wire.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+Verify number of connected SD Wire devices
+    [Documentation]    Test confirms the number of connected SD Wire devices to
+    ...    the DUT platform.
+    ${sd_wire_id_list}=    Get List Of SD Wire Ids
+    ${length_of_sd_wire_id_list}=    Get Length    ${sd_wire_id_list}
+    # this is done to check whether the id of the SD Wire that's connected
+    # to the RTE matches the expected value.
+    Should Be Equal    ${length_of_sd_wire_id_list}    ${SD_WIRES_CONNECTED}
+    FOR    ${index}    IN RANGE    0    ${SD_WIRES_CONNECTED}
+        Should Be Equal    ${sd_wire_id_list[${index}]}    ${SD_WIRE_SERIAL1}
+    END
+
+Flash platform and verify
+    [Documentation]    This test flashes the DUT connected to the RTE through
+    ...    the SD Wire, then attempts to log into it over serial to see whether
+    ...    it works.
+    IF    '${PREV_TEST_STATUS}' == 'FAIL'
+        Fail    'Incorrect number of connected SD Wire devices.'
+    END
+    Variable Should Exist    ${DUT_PASSWORD}
+    # flashing
+    Flash SD Card Via SD Wire    ${FILE_BMAP}    ${FILE_GZ}    ${SD_WIRE_SERIAL1}
+    # telnet
+    Telnet.Set Prompt    :~#
+    Telnet.Read Until    login:
+    Telnet.Write Bare    \n
+    Telnet.Login    root    ${DUT_PASSWORD}
+    ${output}=    Telnet.Execute Command    sh -c "cat /etc/os-release | grep VERSION_ID"
+    ${lines}=    Split To Lines    ${output}
+    ${actual_version}=    Evaluate    "${lines[0]}".split("=")[-1]
+    ${file_gz}=    Evaluate    "${FILE_GZ}".split("/")[-1]
```

Comment:
I do not understand why do we have to split it. If file_gz contains version, then FILE_GZ as well and should be enough to this check? Can you explain?

Reply:
@macpijan Yes. So basically my idea is that this needs to be done to prevent false positives. For example someone can make a directory with the version name, and then accidentally put a file in it that doesn't correspond to the directory name. Then the test would still pass, because the `Should Contain` keyword at the end of the test would return true, because it found the version name in the full path to the file. In order to avoid this false positive, it is best to extract just the file name, and then check against that.

Reply:
Ok, so the correct way to solve it is to use: https://manpages.ubuntu.com/manpages/trusty/man1/basename.1.html
Or even better, look up for something similar in RF libraries. Maybe in this one: https://robotframework.org/robotframework/latest/libraries/OperatingSystem.html

This is very common activity to extract basename from path, no need to invent it once again.

Reply:
@macpijan okay, i used keyword from os library, in [this](https://github.com/Dasharo/open-source-firmware-validation/pull/22/commits/f75318784f351a23bddcf3fea7dce2e612baf6b0) commit